plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/DBPDataSource.java

```diff
@@ -94,4 +94,8 @@ public interface DBPDataSource extends DBSInstanceContainer, DBPContextWithAttri
      */
     void initialize(@NotNull DBRProgressMonitor monitor) throws DBException;
 
+    boolean isReconnecting();

```

Comment:
Can you put these in `DBPDataSourceContainer` instead? It already has `isConnected` and `isConnecting`.

Speaking more broadly, I'd make it return an enum describing the current state. It can only be one of those.

Reply:
Honestly, I don't know. Maybe it's not the correct approach to begin with. Maybe during reconnect data source's `isConnected` must return `false`?